includes/class-amp-theme-support.php

```diff
@@ -1745,6 +1745,30 @@ public static function ensure_required_markup( DOMDocument $dom, $script_handles
 			$amp_scripts[ $missing_script_handle ] = AMP_DOM_Utils::create_node( $dom, 'script', $attrs );
 		}
 
+		// Remove scripts that had already been added but couldn't be detected from output buffering.
+		foreach ( array_diff( array_keys( $amp_scripts ), $script_handles ) as $superfluous_script_handle ) {
```

Comment:
`array_diff( array_keys( $amp_scripts ), $script_handles )` may need to be `array_intersect()`'ed with a list of components that are actually prone to cause validation errors. Scripts like `amp-video` and `amp-form` don't cause any issue when being included. Perhaps this is due to some having built-in status. We need to find out.

In that case, there would be no need for the `amp-dynamic-css-classes` check below either.